lib/node_modules/@stdlib/math/base/special/gammaln/src/main.c

```diff
@@ -0,0 +1,550 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+*
+* ## Notice
+*
+* The following copyright, license, and long comment were part of the original implementation available as part of [FreeBSD]{@link https://svnweb.freebsd.org/base/release/9.3.0/lib/msun/src/e_lgamma_r.c}. The implementation follows the original, but has been modified for JavaScript.
```

Comment:
@gunjjoshi Did you happen to check this implementation against the latest implementation in FreeBSD?

Reply:
I checked the [latest implementation](https://svnweb.freebsd.org/base/release/12.2.0/lib/msun/src/e_lgamma_r.c?revision=367086&view=markup), looks like there are some changes in how they're calculating `sin_pi`. For `gammaln`, even in the [older version](https://svnweb.freebsd.org/base/release/9.3.0/lib/msun/src/e_lgamma_r.c?revision=268523&view=markup), they have used `EXTRACT_WORDS`, and are going by a different manner in the beginning, unlike our `javascript` implementation. We haven't used this way of extracting words and shifting them. This is same in both the versions.